### PR TITLE
Extract Attendee Info Reg Step Form Generation into New Classes

### PR DESCRIPTION
**Pull request by tn3rb**
Before we can switch over to forms generated using the new reg form builder, we need to prepare SPCO to handle the transition. 

This PR extracts the attendee information reg step form generation into new classes that are then loaded as needed. When the new reg form builder is successfully generating data, then we can detect if an event has a form section UUID set and generate the form using a new set of classes based on the new data. The new system should be more more performant due to the dramatic reduction in queries needed to populate the reg form.

### CODE REVIEW

File: core/domain/services/registration/form/CountryOptions.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            : $this->country_model->get_all_active_countries();
+        // start with an empty option
+        $country_options = ['' => ''];
+        if (! empty($countries)) {
+            foreach ($countries as $country) {
+                if ($country instanceof EE_Country) {
+                    $country_options[ $country->ID() ] = $country->name();
+                }
+            }
+        }
```
Comment (manzoorwanijk) [inline core/domain/services/registration/form/CountryOptions.php:104]: 👍

File: core/domain/services/registration/form/LegacyAutoCopyAttendeeInfoForm.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                apply_filters(
+                    'FHEE__EE_SPCO_Reg_Step_Attendee_Information__auto_copy_attendee_info__template_args',
+                    []
+                ),
+                true,
+                true
+            )
+        );
+    }
+}
```
Comment (manzoorwanijk) [inline core/domain/services/registration/form/LegacyAutoCopyAttendeeInfoForm.php:28]: missing EOL

File: core/domain/services/registration/form/LegacyCopyAttendeeInfoForm.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                        'html_id'                 => 'spco-copy-attendee-chk-' . $registration->reg_url_link(),
+                        'html_class'              => 'spco-copy-attendee-chk ee-do-not-validate',
+                        'display_html_label_text' => false,
+                    ]
+                );
+            }
+        }
+        return $copy_attendee_info_inputs;
+    }
+}
```
Comment (manzoorwanijk) [inline core/domain/services/registration/form/LegacyCopyAttendeeInfoForm.php:87]: EOL

**Review by manzoorwanijk — APPROVED** [review]
Looks great 👍

File: core/domain/services/registration/form/LegacyAutoCopyAttendeeInfoForm.php (hunk 2 above)
Comment (tn3rb) [reply core/domain/services/registration/form/LegacyAutoCopyAttendeeInfoForm.php:28]: oops thnx, will fix

File: core/domain/services/registration/form/LegacyCopyAttendeeInfoForm.php (hunk 3 above)
Comment (tn3rb) [reply core/domain/services/registration/form/LegacyCopyAttendeeInfoForm.php:87]: thnx, will fix

**Pull request merged**